Tests/Unit/IoTHubRegistryManagerTests.cs

```diff
@@ -0,0 +1,358 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.Tests.Unit
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using global::LoRaTools;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Shared;
+    using Moq;
+    using Xunit;
+
+    public class IoTHubRegistryManagerTests
+    {
+        private readonly MockRepository mockRepository;
+        private readonly Mock<RegistryManager> mockRegistryManager;
+
+        public IoTHubRegistryManagerTests()
+        {
+            this.mockRepository = new MockRepository(MockBehavior.Strict);
+            this.mockRegistryManager = this.mockRepository.Create<RegistryManager>();
+        }
+
+        private IDeviceRegistryManager CreateManager()
```

Comment:
This should create a IoTHubRegistryManager, since we are unit testing the class.

Consider making the constructor in IoTHubRegistryManager internal so it can be used from here

Reply:
We are here unit testing the ``IoTHubRegistryManager`` class that depends on the ``RegistryManager`` class. So we mock the  ``RegistryManager```?

Reply:
My point is more that the CreateManager method should actually return a IoTHubRegistryManager and not an IDeviceRegistryManager. I agree with mocking the actual RegistryManager class behind it, as you are already doing.

```csharp
private IoTHubRegistryManager CreateManager() =>
  new IoTHubRegistryManager(() => this.mockRegistryManager.Object);
```

Doing this, you can also validate the proper disposal of the mockRegistryManager, by setting up in the constructor the mock as follows
```csharp
this.mockRegistryManager.Protected().Setup("Dispose", ItExpr.Is<bool>(m => true));
```

In addition, you will need to adjust all the this.CreateManager calls to something like the following for the AddConfigurationAsync test method
```csharp
[Fact]
public async Task AddConfigurationAsync()
{
    // Arrange
    using (var manager = this.CreateManager())
    {
        var configuration = new Configuration("testConfiguration");

        this.mockRegistryManager.Setup(c => c.AddConfigurationAsync(It.Is<Configuration>(x => x == configuration)))
            .ReturnsAsync(configuration);

        // Act
        var result = await manager.AddConfigurationAsync(configuration);

        // Assert
        Assert.Equal(configuration, result);
    }
    this.mockRepository.VerifyAll();
}
```

Reply:
Ok Got it !